### ISSUE TITLE
Enabling VLANS on VirtualServer

### ISSUE THREAD

**Issue by faheem556** [F5Networks/k8s-bigip-ctlr#1621]
Is there an option either in AS3 ConfigMap or VirtualServer CRD to enable VLANs on a Virtual Server? I am looking to do a demo of the controller, and there is a requirement to set VLAN. This currently being done with Terraform using the following configuration:

https://registry.terraform.io/providers/F5Networks/bigip/latest/docs/resources/bigip_ltm_virtual_server#vlans

**Comment by trinaths** [comment]
Using AS3 ConfigMap, you can use existing Vlan objects. Refer to example AS3, https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/declarations/miscellaneous.html#virtual-service-allowing-only-specific-vlans

**Comment by mdditt2000** [comment]
@faheem556 adding "allowVlans" to CRD can be done. Add to Virtualserver, Transport or both?

**Comment by faheem556** [comment]
Should be added to both http and tcp connections.

**Comment by mdditt2000** [comment]
Simular to https://github.com/F5Networks/k8s-bigip-ctlr/issues/1592

To allow

```
"allowVlans": [
                "/Common/VL-e-rt-lb"
            ],
```

**Comment by mdditt2000** [comment]
Jira item in the backlog for CONTCNTR-2357. PM requesting for for CIS 2.3

**Comment by mdditt2000** [comment]
@faheem556 adding object to the CRD. However if you want to use a ConfigMap until CIS 2.3 here is a example of the configuration

https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/declarations/miscellaneous.html#virtual-service-allowing-only-specific-vlans

**Comment by faheem556** [comment]
Hi Mark (@mdditt2000), the problem with config maps is that we can't split them #951. We have scores of services and creating a jumbo config-map with IRULE JSON's doesn't attract the team. We are currently using TF and will reevaluate the CIS piece once 2.3 is released. Thank you for the guidance on this.

**Comment by mdditt2000** [comment]
@faheem556 thank you for your feedback. Nice thing about CRD is that CIS will managed the AS3 backend. Also what you might notice is IPAM is coming for the CRDs. Please let me know you team would like a roadmap update etc.

**Comment by mdditt2000** [comment]
@sravyap135 pushed 1 commit.

Allow vlans support for TransportServer CRD

**Comment by trinaths** [comment]
This issue is fixed with CIS v2.3
